packages/fxa-auth-server/lib/luaScripts/touchSessionToken.lua

```diff
@@ -0,0 +1,49 @@
+local REDIS_SESSION_TOKEN_PROPERTIES = {
+  'lastAccessTime',
+  'location',
+  'uaBrowser',
+  'uaBrowserVersion',
+  'uaOS',
+  'uaOSVersion',
+  'uaDeviceType',
+  'uaFormFactor'
+}
+local REDIS_SESSION_TOKEN_LOCATION_PROPERTIES = {
+  'city',
+  'state',
+  'stateCode',
+  'country',
+  'countryCode'
+}
+
+local function packToken(token)
+  local result = {}
+  for i, name in ipairs(REDIS_SESSION_TOKEN_PROPERTIES) do
+    if name == 'location' and type(token[name]) == 'table' then
+      local location = token[name]
+      local r = {}
+      for j, n in ipairs(REDIS_SESSION_TOKEN_LOCATION_PROPERTIES) do
```

Comment:
I wrote a big long comment here about why and then went back up to [this comment](https://github.com/mozilla/fxa/pull/3278/files?w=1#diff-3758ff68a278bb5d275d4003ab6831dcL12) which has answers to all the questions I had.

Reply:
Yeah, I'll port that comment and ref the og PR